test/integration/scheduler/preemption/preemption_test.go

```diff
@@ -274,6 +301,32 @@ func TestPreemption(t *testing.T) {
 			}),
 			preemptedPodIndexes: map[int]struct{}{0: {}},
 		},
+		{
+			name:            "no preemption when PreFilter returns UnschedulableAndUnresolvable",
+			initTokens:      1,
+			preFilterStatus: framework.NewStatus(framework.UnschedulableAndUnresolvable), // the preemption won't happen.
+			existingPods: []*v1.Pod{
+				initPausePod(&testutils.PausePodConfig{
+					Name:      "victim-pod",
+					Namespace: testCtx.NS.Name,
+					Priority:  &lowPriority,
+					Resources: &v1.ResourceRequirements{Requests: v1.ResourceList{
+						v1.ResourceCPU:    *resource.NewMilliQuantity(400, resource.DecimalSI),
+						v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI)},
+					},
+				}),
+			},
+			pod: initPausePod(&testutils.PausePodConfig{
+				Name:      "preemptor-pod",
+				Namespace: testCtx.NS.Name,
+				Priority:  &highPriority,
+				Resources: &v1.ResourceRequirements{Requests: v1.ResourceList{
+					v1.ResourceCPU:    *resource.NewMilliQuantity(300, resource.DecimalSI),
+					v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI)},
+				},
+			}),
+			preemptedPodIndexes: map[int]struct{}{},
```

Comment:
if this is nil, it seems the existing test logic is problematic: (w/o the fix, the test would also pass)

The `else` loop should be employing a "Eventually" semantics; ohterwise it would always pass. In other words, a) the `p` in else block should be obtained from api server and b) we should wait a bit to ensure the preemption doesn't happen. But this is an existing test issue.

```go
// Wait for preemption of pods and make sure the other ones are not preempted.
for i, p := range pods {
	if _, found := test.preemptedPodIndexes[i]; found { ... } else {
		if p.DeletionTimestamp != nil {
			t.Errorf("Didn't expect pod %v to get preempted.", p.Name)
		}
	}
}
```